platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ProgressEvents.groovy

```diff
@@ -160,7 +160,21 @@ class ProgressEvents implements ProgressListener {
                 assert event.result.startTime == startEvent.eventTime
                 assert event.result.endTime == event.eventTime
             } else if (event instanceof TestOutputEvent || event instanceof TestMetadataEvent) {
-                // Do nothing - these are treated differently, their contents are stored separately from the events
+                def descriptor = event.descriptor
+                assert seen.add(descriptor)
+                assert !running.containsKey(descriptor)
+
+                // These are progress events and should have a parent
+                assert descriptor.parent != null
+                // parent should also be running
+                assert running.containsKey(descriptor.parent)
```

Comment:
🤔 I'm not sure this is a good assumption.  Why can't we report metadata for a group or test prior to starting it running?  It might make more sense to gather and record static metadata ahead of starting a test.